core/src/main/scala/kafka/common/ZkNodeChangeNotificationListener.scala

```diff
@@ -97,8 +97,8 @@ class ZkNodeChangeNotificationListener(private val zkUtils: ZkUtils,
             data.map(notificationHandler.processNotification(_)).getOrElse {
```

Comment:
I don't think so. The existing implementations will fail if `null` is passed. Also, the log message below implies that the intent was to log a warning if the data was `null`.